sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableTransactionActionType.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables.models;
+
+import com.azure.data.tables.TableClient;
+
+/**
+ * The type of action to be executed on a {@link TableEntity} in a transactional batch operation.
+ */
+public enum TableTransactionActionType {
+    /**
+     * Add the entity to the table. This is equivalent to {@link TableClient#createEntity(TableEntity)}.
+     */
+    CREATE,
+
+    /**
+     * Upsert the entity in {@link TableEntityUpdateMode#MERGE} mode. This is equivalent to
+     * {@link TableClient#upsertEntity(TableEntity)}.
+     */
+    UPSERT_MERGE,
+
+    /**
+     * Upsert the entity in {@link TableEntityUpdateMode#REPLACE} mode. This is equivalent to
+     * {@link TableClient#upsertEntity(TableEntity)}.
+     */
+    UPSERT_REPLACE,
```

Comment:
In JS we have a single action for Upsert and another one for Update, and pass the update mode as an optional element. Is it possible to do something like that in Java?

```java
new TableTransactionAction(TableTransactionActionType.UPSERT, firstEntity, "Merge")
new TableTransactionAction(TableTransactionActionType.UPSERT, firstEntity, "Replace")
```

Reply:
To be honest, in this case we are following what .NET did here, but after talking with our Java deputy architect we felt like having a constructor that takes a `TableEntity`, `TableTransactionAction`, `TableTransactionActionType` and an `eTag` would make it so that it could only be used for update/upsert actions, restricting its use. In general it just complicates things more.